source/js/front/render.js

```diff
@@ -1,32 +1,27 @@
 class Render {
-	constructor(likedPostsComponents, LikeInstance) {
-		this.components = likedPostsComponents;
-		this.likeInstance = LikeInstance;
-	}
-	getLikedPosts() {
-		const urlParams = new URLSearchParams(window.location.search);
-		const encodedLikedPosts = urlParams.get('liked-posts');
+	constructor(posts, components) {
+		this.components = components;
+		this.posts = posts;
 
-		if (encodedLikedPosts) {
-			const decodedLikedPosts = JSON.parse(atob(encodedLikedPosts));
-			return decodedLikedPosts;
-		} else {
-			const likedPosts = JSON.parse(localStorage.getItem('liked-posts')) || [];
-			return likedPosts;
-		}
+		if (!this.posts || !components) return;
+
+		this.renderComponents();
 	}
 
-	renderComponents(posts) {
+	renderComponents() {
 		const containers = document.querySelectorAll('[js-like-container]');
-		if (posts && posts.length > 0 && containers) {
+		const urlParams = new URLSearchParams(window.location.search);
+
+		if (this.posts && this.posts.length > 0 && containers) {
```

Comment:
Refactor if-statement into separate function with suitable name for readability.